omx/gstomxaudiodec.c

```diff
@@ -252,11 +252,6 @@ gst_omx_audio_dec_change_state (GstElement * element, GstStateChange transition)
     case GST_STATE_CHANGE_PAUSED_TO_PLAYING:
       break;
     case GST_STATE_CHANGE_PAUSED_TO_READY:
-      if (self->dec_in_port)
```

Comment:
Why is this no longer required?

Reply:
This is part of the element tear down fix. Flushing the component at this points causes a omx component state corruption. Removing this allows the component to close properly as well as the gstreamer element.

Reply:
Yes, it seems that the audio codecs don't support flushing ports, but I think it's better to indicate that at this moment the ports are in flushing state and no more data should be processed, so I would add instead the following lines:

```
      /* DSP does not support flush ports */
      self->dec_out_port->flushing = TRUE;
      self->dec_in_port->flushing = TRUE;
```

Reply:
Addressed: 85952ce66a4ccdd9e628f924c0d678b33934f7c5